fabric-message-api-v1/src/client/java/net/fabricmc/fabric/api/client/message/v1/ClientReceiveMessageEvents.java

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.message.v1;
+
+import java.time.Instant;
+
+import com.mojang.authlib.GameProfile;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.network.message.MessageType;
+import net.minecraft.network.message.SignedMessage;
+import net.minecraft.text.Text;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Contains client-side events triggered when receiving messages.
+ */
+public final class ClientReceiveMessageEvents {
+	private ClientReceiveMessageEvents() {
+	}
+
+	/**
+	 * An event triggered when the client receives a chat message,
+	 * which is any message sent by a player. Mods can use this to block the message.
+	 *
+	 * <p>If a listener returned {@code false}, the message will not be displayed,
+	 * the remaining listeners will not be called (if any), and
+	 * {@link #CHAT_CANCELED} will be triggered instead of {@link #CHAT}.
+	 */
+	public static final Event<AllowChat> ALLOW_CHAT = EventFactory.createArrayBacked(AllowChat.class, listeners -> (message, signedMessage, sender, params, receptionTimestamp) -> {
+		for (AllowChat listener : listeners) {
+			if (!listener.allowReceiveChatMessage(message, signedMessage, sender, params, receptionTimestamp)) {
+				return false;
+			}
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event triggered when the client receives a game message,
+	 * which is any message sent by the server.
+	 * Mods can use this to block the message or toggle overlay.
+	 *
+	 * <p>If a listener returned {@code false}, the message will not be displayed,
+	 * the remaining listeners will not be called (if any), and
+	 * {@link #GAME_CANCELED} will be triggered instead of {@link #GAME}.
+	 *
+	 * <p>Overlay is whether the message will be displayed in the action bar.
+	 * To toggle overlay, return false and call
+	 * {@link net.minecraft.client.network.ClientPlayerEntity#sendMessage(Text, boolean) ClientPlayerEntity.sendMessage(message, overlay)}.
+	 */
+	public static final Event<AllowGame> ALLOW_GAME = EventFactory.createArrayBacked(AllowGame.class, listeners -> (message, overlay) -> {
+		for (AllowGame listener : listeners) {
+			if (!listener.allowReceiveGameMessage(message, overlay)) {
+				return false;
+			}
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event triggered when the client receives a chat message,
+	 * which is any message sent by a player. Is not called when
+	 * {@linkplain #ALLOW_CHAT chat messages are blocked}.
+	 * Mods can use this to modify the message.
+	 *
+	 * <p>Secure chat messages modified with this event will be marked with modified indicator.
+	 * Messages already with the modified indicator or game messages are not affected.
+	 * The unmodified message is used in chat reporting.
+	 */
+	public static final Event<Chat> CHAT = EventFactory.createArrayBacked(Chat.class, listeners -> (message, signedMessage, sender, params, receptionTimestamp) -> {
```

Comment:
Should probably be `MODIFY_CHAT` since the message can be modified. (and the function name would be `modifyReceivedChatMessage`).